include/seastar/net/packet.hh

```diff
@@ -225,7 +225,7 @@ public:
     // append temporary_buffer (zero-copy)
     packet(packet&& x, temporary_buffer<char> buf);
     // create from temporary_buffer (zero-copy)
-    packet(temporary_buffer<char> buf);
+    explicit packet(temporary_buffer<char> buf);
     // append deleter
```

Comment:
> 
> Otherwise attempt to put temporary_buffer into output_stream::write()
> implicitly converts the buffer into packet and calls the write(packet)
> overload. Effectively, the write(temporary_buffer) is unused.

I don't understand why. The compiler is supposed to prefer the exact match overload over implicit ones.